src/main/webapp/js/lib/calendar.js

```diff
@@ -49,6 +47,34 @@ const enumDays = {
   DATE_SATURDAY: 6,
 };
 
+/**
+ * The array of colors that a calendar schedule can have.
+ * @type {array}
+ */
+let scheduleColors = [
+  '#ff5f4a',
+  '#b52310',
+  '#f7b32a',
+  '#dba100',
+  '#b3c42f',
+  '#6bd63a',
+  '#3ead0a',
+  '#34d17d',
+  '#009144',
+  '#38d1cc',
+  '#03638c',
+  '#2a2e82',
+  '#491f87',
+  '#8f1088',
+];
+
+/**
+ * A duplicated version of the scheduleColors array. It's used to 'reset' the
+ * scheduleColors array when a new course schedule is requested.
+ * @type {array}
+ */
+const originalColors = [...scheduleColors];
```

Comment:
i think it would make more sense to have `const ORIGINAL_COLORS = ['#ff5f4a', ...]`, and reset `scheduleColors` when it makes sense below. This is kind of "inverting" which one is the constant since you're pulling values from a non-const.